src/webkit/wkPage.ts

```diff
@@ -690,7 +692,9 @@ export class WKPage implements PageDelegate {
   }
 
   async closePage(runBeforeUnload: boolean): Promise<void> {
-    this._pageProxySession.sendMayFail('Target.close', {
+    if (this._recordingVideoFile)
+      await this.stopScreencast();
```

Comment:
do we auto-await screencast stopping on the browser side when popup closes itself?

Reply:
Not yet, the implementation is incomplete. I'd like to first get the events and control methods in place and then iterate on the implementation. One thing we already discovered after looking at the API is that in FF we don't throttle page so would have to add browser events.